protocols/v2/framing-sv2/README.md

```diff
@@ -0,0 +1,60 @@
+](https://github.com/stratum-mining/sv2-spec/blob/main/03-Protocol-Overview.md#32-framing)):
+
+| Protocol Type  | Byte Length | Description |
+|----------------|-------------|-------------|
+| `extension_type` | `U16` | Unique identifier of the extension describing this protocol message. <br><br> Most significant bit (i.e.bit `15`, `0`-indexed, aka `channel_msg`) indicates a message which is specific to a channel, whereas if the most significant bit is unset, the message is to be interpreted by the immediate receiving device. <br><br> Note that the `channel_msg` bit is ignored in the extension lookup, i.e.an `extension_type` of `0x8ABC` is for the same "extension" as `0x0ABC`. <br><br> If the `channel_msg` bit is set, the first four bytes of the payload field is a `U32` representing the `channel_id` this message is destined for (these bytes are repeated in the message framing descriptions below). <br><br> Note that for the Job Declaration and Template Distribution Protocols the `channel_msg` bit is always unset. |
+| `msg_type` | `U8` | Unique identifier of the extension describing this protocol message. |
+| `msg_length` | `U24` | Length of the protocol message, not including this header. |
+| `payload` | `BYTES` | Message-specific payload of length `msg_length`. If the MSB in `extension_type` (the `channel_msg` bit) is set the first four bytes are defined as a `U32` `"channel_id"`, though this definition is repeated in the message definitions below and these 4 bytes are included in `msg_length`. |
+
+## Main Components
+
+- **Header**: Defines the 6-byte Sv2 message header with information about the message payload,
+  including its extension type, if it is associated with a specific mining channel, the type of
+  message (e.g. `SetupConnection`, `NewMiningJob`, etc.) and the payload length.
+- **Sv2 Framing**: Use for serializing Sv2 messages.
+- **Noise Handshake Framing**: Use for serializing Noise protocol handshake messages.
+
+## Usage
+
+To include this crate in your project, run:
+
+```bash
+cargo add framing_sv2
+```
+
+This crate can be built with the following feature flags:
+
+- `with_buffer_pool`: Enables buffer pooling for more efficient memory management.
+- `with_serde`: builds [`binary_sv2`](https://crates.io/crates/binary_sv2) and
+  [`buffer_sv2`](https://crates.io/crates/buffer_sv2) crates with `serde`-based encoding and
+  decoding. Note that this feature flag is only used for the Message Generator, and deprecated
+  for any other kind of usage. It will likely be fully deprecated in the future.
+
+### Examples
+
+This crate provides two examples demonstrating how to serialize and deserialize Sv2 and Noise
+handshake message frames:
+
+1. **[Sv2 Frame](https://github.com/stratum-mining/stratum/blob/main/protocols/v2/framing-sv2/examples/sv2_frame.rs)**:
+   Encode and decode standard Sv2 frames, detailing the message serialization and transmission
+   process for unencrypted communications.
```

Comment:
there is a lot of conceptual overlap between encode/decode vs serialize/deserialize

to be fully honest, I often get confused and also conflate the two

but for the sake of clarity, we should reserve the usage of encode/decode terminology for the `codec_sv2` crate.

Reply:
also, we should avoid bringing up "unencrypted" here... sure, the example is unencrypted, but from a conceptual standpoint, a `Sv2Frame` can be serialized into bytes just like on this example, and later be transmitted over an encrypted channel

so drawing attention to "unencrypted" here is misleading